presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -362,6 +362,12 @@ public PlanWithProperties visitExchange(ExchangeNode node, StreamPreferredProper
         {
             checkArgument(node.getScope() != LOCAL, "AddLocalExchanges can not process a plan containing a local exchange");
             // this node changes the input organization completely, so we do not pass through parent preferences
+            if (node.isOrderSensitive()) {
+                return planAndEnforceChildren(
+                        node,
+                        parentPreferences.withoutPreference().withDefaultParallelism(session),
```

Comment:
In `LocalExecutionPlanner` when isOrderSensitive is set, the driver instance count is set to 1. Why is this passing through preferences that are somewhat different?